src/ray/raylet/node_manager.cc

```diff
@@ -323,12 +323,22 @@ NodeManager::NodeManager(instrumented_io_context &io_service, const NodeID &self
   }
 
   auto options = AgentManager::Options({self_node_id, agent_command_line});
-  agent_manager_.reset(
-      new AgentManager(std::move(options),
-                       /*delay_executor=*/
-                       [this](std::function<void()> task, uint32_t delay_ms) {
-                         return execute_after(io_service_, task, delay_ms);
-                       }));
+  agent_manager_ = std::make_shared<AgentManager>(
+      std::move(options),
+      /*delay_executor=*/
+      [this](std::function<void()> task, uint32_t delay_ms) {
+        return execute_after(io_service_, task, delay_ms);
+      },
+      /*runtime_env_agent_factory=*/
+      [this](const std::string &ip_address, int port) {
+        if (ip_address.empty() || port == 0) {
+          RAY_LOG(INFO) << "client nullptr";
```

Comment:
1) why will the ip or the port be empty? Is this only for tests?
2) Make the log clearer.

Reply:
Maybe no ip or port in agent register message.

Reply:
Or in tests.

Reply:
Maybe I should use RAY_CHECK instead.